### PR TITLE
GH#53054 modules/update-upgrading-cli: Use 'oc adm upgrade' for current status

### PR DESCRIPTION
**Pull request by obrown1205**
Version(s):
4.10+

Issue:
Resolving conflicts from https://github.com/openshift/openshift-docs/pull/53054

Link to docs preview:
https://55005--docspreview.netlify.app/openshift-enterprise/latest/updating/updating-cluster-cli.html#update-upgrading-cli_updating-cluster-cli

QE review:
- [x] QE has approved this change.

Additional information:

### CODE REVIEW

File: modules/update-upgrading-cli.adoc [hunk 1]
```diff
@@ -46,12 +46,12 @@ $ oc adm upgrade
 ----
 +
 .Example output
-[source,terminal]
+[source,terminal,subs="attributes+"]
 ----
 Cluster version is 4.9.23
```
Comment (wking) [inline modules/update-upgrading-cli.adoc:51]: Do you want `4.{product-version}.23` as the dummy version here too?

File: modules/update-upgrading-cli.adoc [hunk 2]
```diff
@@ -46,12 +46,12 @@ $ oc adm upgrade
...
 
 Upstream is unset, so the cluster will use an appropriate default.
-Channel: stable-4.10 (available channels: candidate-4.10, candidate-4.9, fast-4.10, fast-4.9, stable-4.10, stable-4.9)
+Channel: stable-{product-version} (available channels: candidate-{product-version}, fast-{product-version}, stable-{product-version})
```
Comment (wking) [inline modules/update-upgrading-cli.adoc:54]: We're currently including `eus-...` in our channel recommendations below, so we probably want to include `eus-...` somehow in this line.  But we probably don't want to use `eus-{product-version}`, because only 4.even are EUS (e.g. both 4.11 and 4.12 docs builds would want to use `eus-4.12`).  Not sure if it's worth defining `{eus-inclusion-version}` or something?

File: modules/update-upgrading-cli.adoc [hunk 3]
```diff
@@ -46,12 +32,12 @@ $ oc adm upgrade
 ----
 +
 .Example output
-[source,terminal]
+[source,terminal,subs="attributes+"]
```
Comment (wking) [inline modules/update-upgrading-cli.adoc:35]: nit: you may no longer need `attributes` now that you no longer use `{product-version}` in the example output?

File: modules/update-upgrading-cli.adoc [hunk 4]
```diff
@@ -75,19 +61,21 @@ VERSION IMAGE
 4.10.12 quay.io/openshift-release-dev/ocp-release@sha256:f77f4f75c1e1a4ddd0a0355f298a834db3473fd9ca473235013e9419d1df16db
 4.10.13 quay.io/openshift-release-dev/ocp-release@sha256:4f516616baed3cf84585e753359f7ef2153ae139c2e80e0191902fbd073c4143
 ----
+[NOTE]
+*For details and information on how to perform an `EUS-to-EUS` channel upgrade, please refer to the link:https://docs.openshift.com/container-platform/4.12/updating/preparing-eus-eus-upgrade.html[Preparing to perform an EUS-to-EUS upgrade] page.
```
Comment (wking) [inline modules/update-upgrading-cli.adoc:65]: is there a way for us to link here without  pinning 4.12?

Comment (obrown1205) [reply modules/update-upgrading-cli.adoc:65]: I'll look into this :)

File: modules/update-upgrading-cli.adoc (hunk 3 above)
Comment (obrown1205) [reply modules/update-upgrading-cli.adoc:35]: ah - good catch!

File: modules/update-upgrading-cli.adoc [hunk 5]
```diff
@@ -18,25 +18,11 @@ of the Customer Portal.
 
 * Install the OpenShift CLI (`oc`) that matches the version for your updated version.
 * Log in to the cluster as user with `cluster-admin` privileges.
-* Install the `jq` package.
+
 * Pause all `MachineHealthCheck` resources.
```
Comment (LalatenduMohanty) [inline modules/update-upgrading-cli.adoc:22]: May be we can handle this in another PR but we should give the steps to pause all `MachineHealthCheck` resources.

File: modules/update-upgrading-cli.adoc [hunk 6]
```diff
@@ -75,19 +61,21 @@ VERSION IMAGE
 4.10.12 quay.io/openshift-release-dev/ocp-release@sha256:f77f4f75c1e1a4ddd0a0355f298a834db3473fd9ca473235013e9419d1df16db
 4.10.13 quay.io/openshift-release-dev/ocp-release@sha256:4f516616baed3cf84585e753359f7ef2153ae139c2e80e0191902fbd073c4143
 ----
+[NOTE]
+*For details and information on how to perform an `EUS-to-EUS` channel upgrade, please refer to the link:https://docs.openshift.com/container-platform/4.12/updating/preparing-eus-eus-upgrade.html[Preparing to perform an EUS-to-EUS upgrade] page.
 
-. Based on your organization requirements, set the upgrade channel to `stable-{product-version}`, `fast-{product-version}`, or `eus-{product-version}`:
+. Based on your organization requirements, set the upgrade channel to `stable-4.9`, `fast-4.9`, or `eus-*`:
```
Comment (LalatenduMohanty) [inline modules/update-upgrading-cli.adoc:67]: We should say `Based on your organization requirements, set the appropriate upgrade channel. For example: `stable-4.9`, `fast-4.9`, or `eus-*`. Refer to the channel documentation (link to the channel docs[1]) for more information about channels. 
[1] https://docs.openshift.com/container-platform/4.12/updating/understanding-upgrade-channels-release.html

**Review by LalatenduMohanty — CHANGES_REQUESTED** [review]

File: modules/update-upgrading-cli.adoc (hunk 4 above)
Comment (obrown1205) [reply modules/update-upgrading-cli.adoc:65]: @wking @LalatenduMohanty  
After looking into this, I actually believe it's important to keep this up-to-date, as it still includes information for all of our still-supported versions and is maintained within our team. After reading through it, I don't see any reasons that the pining of 4.12 documentation should confuse our 4.9 customers. A couple of points stick out to me in the 4.12 version. For example:

- "You must be running at least OpenShift Container Platform 4.8.14 before starting the EUS-to-EUS update procedure. If you do not meet this minimum requirement, update to a later 4.8.z before attempting the EUS-to-EUS update."

- "Support for RHEL7 workers was removed in OpenShift Container Platform 4.10 and replaced with RHEL8 workers, therefore EUS-to-EUS updates are not available for clusters with RHEL7 workers."

- "Node components are not updated to OpenShift Container Platform 4.9. Do not expect all features and bugs fixed in OpenShift Container Platform 4.9 to be made available until you complete the update to OpenShift Container Platform 4.10 and enable all MachineConfigPools to update."

Are there any specific concerns we should address with 4.12?

Comment (skopacz1) [reply modules/update-upgrading-cli.adoc:65]: @wking yes, kind of, though it's a little ugly and clunky. 

What @obrown1205 could do is not link the page directly in this note, since it's in a module and can't use a version agnostic xref, but instead just refer to the page title in italics here. 

Then, in the [Additional Resources section right underneath this procedure](https://docs.openshift.com/container-platform/4.12/updating/updating-cluster-cli.html#update-upgrading-cli_updating-cluster-cli), where there's already a link to the Understanding Upgrade Channels page, she could add another xref to the EUS to EUS page (the additional resources section is located in the assembly for the entire page, not a module, so xrefs are allowed)

File: modules/update-upgrading-cli.adoc (hunk 5 above)
Comment (obrown1205) [reply modules/update-upgrading-cli.adoc:22]: Actually - the instructions are on the page: https://docs.openshift.com/container-platform/4.12/updating/updating-cluster-cli.html#machine-health-checks-pausing_updating-cluster-cli (with the anchor at the end for specifics). I'll go ahead and delete this as a prereq.

Comment (obrown1205) [reply modules/update-upgrading-cli.adoc:22]: After more digging - looks like it's covered here: https://docs.openshift.com/container-platform/4.12/updating/updating-cluster-cli.html#machine-health-checks-pausing_updating-cluster-cli so we should be set with the prereq as is

Comment (wking) [reply modules/update-upgrading-cli.adoc:22]: I am not actually clear why we  recommend pausing MHCs before updating.  But I haven't had time to hunt around in  the pull request that originally added this.  I'd like to get the need clarified before we sink time into polishing the request, because we don't want to waste polishing time on something that's going away.

File: modules/update-upgrading-cli.adoc [hunk 7]
```diff
@@ -48,10 +34,10 @@ $ oc adm upgrade
 .Example output
 [source,terminal]
 ----
-Cluster version is 4.9.23
+Cluster version is 4.{product-version}.23
 
 Upstream is unset, so the cluster will use an appropriate default.
-Channel: stable-4.10 (available channels: candidate-4.10, candidate-4.9, fast-4.10, fast-4.9, stable-4.10, stable-4.9)
+Channel: stable-4.9.23 (available channels: candidate-4.9.23, fast-4.9.23, stable-4.9.23, eus-*)
```
Comment (wking) [inline modules/update-upgrading-cli.adoc:40]: Channels are just 4.y, like `stable-4.9`, not `stable-4.9.23`, and similar for the other channels.  We'll also need to use a 4.10 channel  to get the 4.10.13, etc. examples in the following `Recommended updates` section, so we might want to stick with `stable-4.10` as the selected channel  here.  Or we can move it to `stable-4.9` and then drop all the 4.10.z targets below.

Comment (obrown1205) [reply modules/update-upgrading-cli.adoc:40]: @wking would this mean adjusting stable-4.9 to stable-4.10 and removing the 4.9.x examples from the example output?

File: modules/update-upgrading-cli.adoc (hunk 5 above)
Comment (obrown1205) [reply modules/update-upgrading-cli.adoc:22]: @wking According to the page for pausing MHC's:

"During the upgrade process, nodes in the cluster might become temporarily unavailable. In the case of worker nodes, the machine health check might identify such nodes as unhealthy and reboot them. To avoid rebooting such nodes, pause all the MachineHealthCheck resources before updating the cluster."

I suspect this is still relevant, but I could be wrong. Thoughts?

Comment (obrown1205) [reply modules/update-upgrading-cli.adoc:22]: Team decision to address MHC's in a new PR for the future. JIRA story here: https://issues.redhat.com/browse/OTA-873

File: modules/update-upgrading-cli.adoc [hunk 8]
```diff
@@ -75,19 +61,22 @@ VERSION IMAGE
 4.10.12 quay.io/openshift-release-dev/ocp-release@sha256:f77f4f75c1e1a4ddd0a0355f298a834db3473fd9ca473235013e9419d1df16db
 4.10.13 quay.io/openshift-release-dev/ocp-release@sha256:4f516616baed3cf84585e753359f7ef2153ae139c2e80e0191902fbd073c4143
 ----
+[NOTE]
+*For details and information on how to perform an `EUS-to-EUS` channel upgrade, please refer to the 
+_Preparing to perform an EUS-to-EUS upgrade page_, listed in the Additional resources section.
```
Comment (mburke5678) [inline modules/update-upgrading-cli.adoc:66]: Note formatting:

[NOTE]
====
For details and information on how to perform an `EUS-to-EUS` channel upgrade, please refer to the 
_Preparing to perform an EUS-to-EUS upgrade page_, listed in the Additional resources section.
====

Comment (mburke5678) [reply modules/update-upgrading-cli.adoc:66]: s/_Preparing to perform an EUS-to-EUS upgrade page_/_Preparing to perform an EUS-to-EUS upgrade_ page

**Comment by mburke5678** [comment]
@obrown1205 A few nits. Otherwise LGTM

File: modules/update-upgrading-cli.adoc [hunk 9]
```diff
@@ -51,7 +37,7 @@ $ oc adm upgrade
 Cluster version is 4.9.23
 
 Upstream is unset, so the cluster will use an appropriate default.
-Channel: stable-4.10 (available channels: candidate-4.10, candidate-4.9, fast-4.10, fast-4.9, stable-4.10, stable-4.9)
+Channel: stable-4.10 (available channels: candidate-4.10, candidate-4.9, fast-4.10, fast-4.9, stable-4.10, stable-4.9, eus-*)
```
Comment (shellyyang1989) [inline modules/update-upgrading-cli.adoc:40]: eus-*  -> eus-4.10

File: modules/update-upgrading-cli.adoc [hunk 10]
```diff
@@ -75,19 +61,22 @@ VERSION IMAGE
...
 ----
 +
-For example, to set the channel to `stable-{product-version}`:
+For example, to set the channel to `stable-4.9`:
 +
-[source,terminal,subs="attributes+"]
+[source,terminal]
 ----
-$ oc adm upgrade channel stable-{product-version}
+$ oc adm upgrade channel stable-4.9
```
Comment (shellyyang1989) [inline modules/update-upgrading-cli.adoc:79]: The channel version in the examples looks not that good to me. The default channel on a 4.9 cluster is stable-4.9. I'd prefer to have stable-4.9 as the example channel in procedure#1 and switch to stable-4.10 in the step.

Comment (obrown1205) [reply modules/update-upgrading-cli.adoc:79]: Hi @shellyyang1989 ! Sounds good :) I assume this also changes the command 
`$ oc adm upgrade channel stable-4.9` to `$ oc adm upgrade channel stable-4.10` on line 82?

File: modules/update-upgrading-cli.adoc (hunk 8 above)
Comment (obrown1205) [reply modules/update-upgrading-cli.adoc:66]: Thank you! Adjusted both :)

File: modules/update-upgrading-cli.adoc [hunk 11]
```diff
@@ -116,78 +99,30 @@ $ oc adm upgrade --to=<version> <1>
...
-. If you are upgrading your cluster to the next minor version, like version 4.y to 4.(y+1), it is recommended to confirm your nodes are upgraded before deploying workloads that rely on a new feature:
+Cluster version is <version> 
+
+Upstream is unset, so the cluster will use an appropriate default.
+Channel: stable-4.10 (available channels: candidate-4.10, candidate-4.11, eus-4.10, fast-4.10, fast-4.11, stable-4.10) 
+
+No updates available. You may force an upgrade to a specific release image, but doing so might not be supported and might result in downtime or data loss.
+----
++
+. If you are upgrading your cluster to the next minor version, like version X.y to X.(y+1), it is recommended to confirm your nodes are upgraded before deploying workloads that rely on a new feature:
```
Comment (mburke5678) [inline modules/update-upgrading-cli.adoc:125]: Per the ISG:  s/like/such as

s/confirm/confirm that

File: modules/update-upgrading-cli.adoc [hunk 12]
```diff
@@ -64,30 +50,27 @@ VERSION IMAGE
...
-4.10.13 quay.io/openshift-release-dev/ocp-release@sha256:4f516616baed3cf84585e753359f7ef2153ae139c2e80e0191902fbd073c4143
-----
-
-. Based on your organization requirements, set the upgrade channel to `stable-{product-version}`, `fast-{product-version}`, or `eus-{product-version}`:
+
+----
++
+[NOTE]
+====
+*For details and information on how to perform an `EUS-to-EUS` channel upgrade, please refer to the
```
Comment (mburke5678) [inline modules/update-upgrading-cli.adoc:58]: Don't need the asterisk here.
